src/App.tsx

```diff
@@ -14,11 +17,113 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type Props = {};
+type State = {
+  isStarted: boolean,
+  goodsList: string[],
+  visibleGoods: string[],
+  isReversed: boolean,
+};
+
+class App extends React.Component<Props, State> {
+  state = {
+    isStarted: false,
+    goodsList: goodsFromServer,
+    visibleGoods: goodsFromServer,
```

Comment:
This one isn't incorrect, however, I would rather create state properties indicating the way I need to render my list. For instance, `isReversed` and its further usage: if the flag is up I reverse goods list, otherwise I skip reversing

Reply:
I've done it so, thus you could check my code and decide if you like or not

Reply:
By doing this away, you avoid at least repeating yourself
```
  goodsList: goodsFromServer,
  visibleGoods: goodsFromServer,
```